mlflow/tracking/_model_registry/fluent.py

```diff
@@ -107,27 +116,55 @@ def _register_model(
         else:
             raise e
 
-    run_id = None
+    run_id, model_id = None, None
     source = model_uri
     if RunsArtifactRepository.is_runs_uri(model_uri):
-        source = RunsArtifactRepository.get_underlying_uri(model_uri)
-        (run_id, _) = RunsArtifactRepository.parse_runs_uri(model_uri)
+        # If the uri is of the form runs:/...
+        (run_id, artifact_path) = RunsArtifactRepository.parse_runs_uri(model_uri)
+        runs_artifact_repo = RunsArtifactRepository(model_uri)
+        if runs_artifact_repo._is_directory(artifact_path):
+            # First check if run has artifact at artifact_path,
+            # if so use the run's artifact location as source
+            source = RunsArtifactRepository.get_underlying_uri(model_uri)
+        else:
+            # Otherwise check if there's a logged model with
+            # name artifact_path and source_run_id run_id
+            run = client.get_run(run_id)
+            logged_models = _get_logged_models_from_run(run, artifact_path)
+            if not logged_models:
+                raise MlflowException(
+                    f"Unable to find a logged_model with artifact_path {artifact_path} "
+                    + f"under run {run_id}",
+                    error_code=ErrorCode.Name(NOT_FOUND),
+                )
+            # If there are multiple such logged models, get the one logged at the largest step
+            model_id_to_step = {m_o.model_id: m_o.step for m_o in run.outputs.model_outputs}
+            model_id = sorted(
+                logged_models, key=lambda lm: model_id_to_step[lm.model_id], reverse=True
+            )[0].model_id
```

Comment:
```suggestion
            model_id = max(
                logged_models, key=lambda lm: model_id_to_step[lm.model_id], reverse=True
            ).model_id
```

`max` or `min` should work.